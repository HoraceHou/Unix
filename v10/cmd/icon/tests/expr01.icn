record array(a,b,c,d,e,f,g)

procedure dummy(u,v,x,y,z)
   suspend u | v
   return x
end

procedure main()
   &trace := -1
   dummy{1,2,3,4}
   dummy{}
   dummy(image{1,2,3,4})
   dummy(put{1,2,3,4})
   dummy("*"{1,2,3,4})
   dummy(image("|||"([],[])))
   dummy("+"(1,2))
   dummy("+"(1))
   dummy("image"(image(image)))
   dummy("x"(1))
end
