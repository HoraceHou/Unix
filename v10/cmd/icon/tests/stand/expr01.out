expr01.icn: 10	| dummy([co-expression,co-expression,co-expression,co-expression],&null,&null,&null,&null)
expr01.icn: 4	| dummy suspended [co-expression,co-expression,co-expression,co-expression]
expr01.icn: 11	| dummy([],&null,&null,&null,&null)
expr01.icn: 4	| dummy suspended []
expr01.icn: 12	| dummy("list(4)",&null,&null,&null,&null)
expr01.icn: 4	| dummy suspended "list(4)"
expr01.icn: 13	| dummy([co-expression,co-expression,co-expression,co-expression,],&null,&null,&null,&null)
expr01.icn: 4	| dummy suspended [co-expression,co-expression,co-expression,co-expression,]
expr01.icn: 14	| dummy(4,&null,&null,&null,&null)
expr01.icn: 4	| dummy suspended 4
expr01.icn: 15	| dummy("list(0)",&null,&null,&null,&null)
expr01.icn: 4	| dummy suspended "list(0)"
expr01.icn: 16	| dummy(3,&null,&null,&null,&null)
expr01.icn: 4	| dummy suspended 3
expr01.icn: 17	| dummy(1,&null,&null,&null,&null)
expr01.icn: 4	| dummy suspended 1
expr01.icn: 18	| dummy("\"function image\"",&null,&null,&null,&null)
expr01.icn: 4	| dummy suspended "\"function image\""
Run-time error 106 at line 19 in expr01.icn
procedure or integer expected
offending value: "x"
